### PR TITLE
go plugin: remove confusing importpath message

### PR DESCRIPTION
**Pull request by Nissaar**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: go.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
+bugbu# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
```
Comment (kyrofa) [inline go.py]: This probably shouldn't be here :wink: .

**Comment by kyrofa** [comment]
Also, your pull request and title (and commit message, from which the former are derived) should probably discuss why the change was needed and what bug it's fixing. The current title doesn't mean much.

File: go.py (hunk 1 above)
Comment (kyrofa) [reply go.py]: Please delete this file.

**Comment by kyrofa** [comment]
Alright, now this is ready for a good review! Thanks for the changes so far, @Nissaar. I have a question. I know this is fixing LP: [#1613980](https://bugs.launchpad.net/snapcraft/+bug/1613980), however, that bug contains the following paragraph:

> The "This is not needed" and printing systematically a warning is confusing. As for most of projects, we don't need go-importpath, I would suggest removing the warning.

I read that as suggesting that we remove the "please consider" warning, not the "this is not needed" documentation. Did you read it the other way? I'm afraid I don't remember why we added the warning in the first place, so I'd like to get @elopio (who marked the bug as confirmed) and @sergiusens's input on this as well.

**Comment by sergiusens** [comment]
The "This is not needed" should probably be better read as "in some cases this is not required", the reason it is not required is to not break backwards compatibility and ideally should be set. Maybe this should become a deprecation message in the sense of it being deprecated to craft a snapcraft.yaml with the go plugin without this entry.

**Comment by kyrofa** [comment]
So @sergiusens is this PR good as-is? Or would you like to see it also upgrade the warning to a deprecation?

File: snapcraft/plugins/go.py [hunk 2]
```diff
@@ -30,13 +30,12 @@
       not be useful either.
       If the package is a part of the go-importpath the local package
       corresponding to those sources will be used.
-
-    - go-importpath:
+      
+      - go-importpath:
```
Comment (kyrofa) [inline snapcraft/plugins/go.py]: Please fix the extra spaces and indentation here.

**Comment by kyrofa** [comment]
Closing due to inactivity. Please re-propose once you've addressed the issues.

**Pull request closed without merging**